### PR TITLE
Rework internal vm agent logs to include function and task id

### PR DESCRIPTION
**Pull request by austinvazquez**
*Issue #, if available:*
N/A

*Description of changes:*
Tracking a production issue which has the agent process dying as a result of a panic. This change is to enhance the agent logs by logging the request function name and task ID for each panic handler.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

Signed-off-by: Austin Vazquez <macedonv@amazon.com>

### CODE REVIEW

File: agent/service.go [hunk 1]
```diff
@@ -159,8 +166,7 @@ func (ts *TaskService) Create(requestCtx context.Context, req *taskAPI.CreateTas
 		return nil, fmt.Errorf("invalid task and/or exec ID: %w", err)
 	}
 
-	logger := log.G(requestCtx).WithField("TaskID", taskID).WithField("ExecID", execID)
-	logger.Info("create")
+	logger.Debug("create")
```
Comment (austinvazquez) [inline agent/service.go:169]: `Debug` to be consistent with other handlers.

File: agent/service.go [hunk 2]
```diff
@@ -150,7 +151,13 @@ func (ts *TaskService) doCleanup(taskExecID string) error {
 
 // Create creates a new initial process and container using runc
 func (ts *TaskService) Create(requestCtx context.Context, req *taskAPI.CreateTaskRequest) (_ *taskAPI.CreateTaskResponse, err error) {
-	defer logPanicAndDie(log.G(requestCtx))
+	logger := log.G(requestCtx).WithFields(logrus.Fields{
+		"name":    "Create",
```
Comment (austinvazquez) [inline agent/service.go:155]: Open to feedback on the key name here.

File: agent/service.go [hunk 3]
```diff
@@ -525,107 +572,139 @@ func (ts *TaskService) Exec(requestCtx context.Context, req *taskAPI.ExecProcess
...
 		return nil, err
 	}
 
-	log.G(requestCtx).WithFields(logrus.Fields{
+	logger.WithFields(logrus.Fields{
 		"shim_pid": resp.ShimPid,
 		"task_pid": resp.TaskPid,
 		"version":  resp.Version,
-	}).Error("connect succeeded")
+	}).Debug("connect succeeded")
```
Comment (austinvazquez) [inline agent/service.go:690]: Don't error on success.

File: agent/service.go (hunk 2 above)
Comment (henry118) [reply agent/service.go:155]: Did you remove the log panic call?

Comment (austinvazquez) [reply agent/service.go:155]: It is line 159 now, basically setting up the logger with fields so if a panic occurs we can see what function and container caused the panic.

**Review by henry118 — APPROVED** [review]

**Review by swagatbora90 — APPROVED** [review]

**Pull request merged**